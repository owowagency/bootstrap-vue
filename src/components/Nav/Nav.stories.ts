import Nav from '.';
import NavItem from '@/components/NavItem';
import NavLink from '@/components/NavLink';

export default {
    title: 'Components/Nav',
    component: Nav,
};

const template = (args) => ({
    components: {Nav, NavItem, NavLink},
    setup: () => ({args}),
    template: `
        <Nav v-bind="args">
            <NavItem>
                <NavLink active>
                    Active
                </NavLink>
            </NavItem>

            <NavItem>
                <NavLink>
                    Link
                </NavLink>
            </NavItem>

            <NavItem>
                <NavLink>
                    Link
                </NavLink>
            </NavItem>

            <NavItem disabled>
                <NavLink disabled>
                    Disabled
                </NavLink>
            </NavItem>
        </Nav>
    `,
});

export const Default = template.bind({});

export const Vertical = template.bind({});

Vertical.args = {
    vertical: true,
};

export const Tabs = template.bind({});

Tabs.args = {
    tabs: true,
};

export const Pills = template.bind({});

Pills.args = {
    pills: true,
};

export const Fill = template.bind({});

Fill.args = {
    fill: true,
};

export const Justified = template.bind({});

Justified.args = {
    justified: true,
};
