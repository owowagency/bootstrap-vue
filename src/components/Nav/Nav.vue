<template>
    <nav
        class="nav"
        :class="classes"
    >
        <!-- @slot Displays the content of the nav -->
        <slot />
    </nav>
</template>

<script lang="ts" setup>
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';

const props = defineProps({
    /**
     * Defines if the `nav-fill` class should be used
     */
    fill: {
        type: Boolean,
        default: false,
    },
    /**
     * Defines if the `nav-justified` class should be used
     */
    justified: {
        type: Boolean,
        default: false,
    },
    /**
     * Defines if the `nav-pills` class should be used
     */
    pills: {
        type: Boolean,
        default: false,
    },
    /**
     * Defines if the `nav-tabs` class should be used
     */
    tabs: {
        type: Boolean,
        default: false,
    },
    /**
     * Defines if the `flex-column` class should be used
     * *Displays items vertically*
     */
    vertical: {
        type: Boolean,
        default: false,
    },
});

const {classes} = useClasses(computed(() => [
    props.fill ? 'nav-fill' : undefined,
    props.justified ? 'nav-justified' : undefined,
    props.pills ? 'nav-pills' : undefined,
    props.tabs ? 'nav-tabs' : undefined,
    props.vertical ? 'flex-column' : undefined,
]));
</script>

<docs>
```vue
<Nav fill tabs>
    <NavItem>Home</NavItem>
    <NavItem>Docs</NavItem>
    <NavItem>About</NavItem>
</Nav>
```
</docs>
