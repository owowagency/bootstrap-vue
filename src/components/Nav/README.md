# Nav

```vue
<Nav fill tabs>
    <NavItem>Home</NavItem>
    <NavItem>Docs</NavItem>
    <NavItem>About</NavItem>
</Nav>
```

## Props

| Prop name | Description                                                                       | Type    | Values | Default |
| --------- | --------------------------------------------------------------------------------- | ------- | ------ | ------- |
| fill      | Defines if the `nav-fill` class should be used                                    | boolean | -      | false   |
| justified | Defines if the `nav-justified` class should be used                               | boolean | -      | false   |
| pills     | Defines if the `nav-pills` class should be used                                   | boolean | -      | false   |
| tabs      | Defines if the `nav-tabs` class should be used                                    | boolean | -      | false   |
| vertical  | Defines if the `flex-column` class should be used<br/>_Displays items vertically_ | boolean | -      | false   |

## Slots

| Name    | Description                     | Bindings |
| ------- | ------------------------------- | -------- |
| default | Displays the content of the nav |          |
