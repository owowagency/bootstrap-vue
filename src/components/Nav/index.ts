import Nav from './Nav.vue';

export default Nav;
