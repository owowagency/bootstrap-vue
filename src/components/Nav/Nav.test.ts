import Nav from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(Nav);

    componentSlotRenderTest(Nav);

    componentWrapperClassTest(Nav, {fill: true}, 'nav-fill');

    componentWrapperClassTest(Nav, {justified: true}, 'nav-justified');

    componentWrapperClassTest(Nav, {pills: true}, 'nav-pills');

    componentWrapperClassTest(Nav, {tabs: true}, 'nav-tabs');

    componentWrapperClassTest(Nav, {vertical: true}, 'flex-column');
});
