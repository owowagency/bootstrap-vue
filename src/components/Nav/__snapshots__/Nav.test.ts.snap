// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<nav class=\\"nav\\">
  <!-- @slot Displays the content of the nav -->
</nav>"
`;
